python/lsst/ts/Dome/dome_csc.py

```diff
@@ -156,32 +150,36 @@ async def handle_summary_state(self):
         components (or the mock_controller) when needed.
         """
         self.log.info("handle_summary_state")
-        # TODO It should be possible to always connect and not just in DISABLED or ENABLED state.
         if self.disabled_or_enabled:
             if not self.connected:
                 await self.connect()
         else:
             await self.disconnect()
 
-    async def read(self):
-        """Utility function to read a string from the reader and unmarshal it.
+    async def write_then_read_reply(self, cmd):
+        """Write the cmd string appended with a newline character and then read the reply to the command.
+
+        Parameters
+        ----------
+        cmd: `dict`
+            A dict compatible with yaml.safe_dump containing a command and possibly its parameters. It
+            should take the form {"cmd":\n "param1": value1\n "param2": value2\n} with zero, one or more
+            parameters.
```

Comment:
Thanks for those suggestions. As ma matter of fact, the doc string is incorrect. cmd is a string with the specified format and the \n characters are mandatory otherwise yaml will choke on it. I'll rewrite the doc string once more.

Reply:
Sorry, my bad. It is a dict and I reformatted the description to match a Python dict.